doc/userdoc/nest_server.rst

```diff
@@ -0,0 +1,649 @@
+``pip``::
+
+  pip3 install RestrictedPython uwsgi flask flask-cors
+
+or by installing the full NEST development environment in case you
+prefer using ``conda``::
+
+  conda env create -f extras/conda-nest-simulator-dev.yml
+  conda acvitate nest-simulator
+
+As an alternative to a native installation, NEST Server is available
+from the NEST Docker image. Please check out the corresponding
+:ref:`installation instructions <docker_vm_install>` for more details.
+
+Run NEST Server
+~~~~~~~~~~~~~~~~~~~~~
+
+All NEST Server operations are managed using the ``nest-server``
+command that can either be run directly::
+
+  nest-server start
+
+or supplied to the execution command line for running the Docker
+container::
+
+  docker run -it --rm -e LOCAL_USER_ID=`id -u $USER` -p 5000:5000 nestsim/nest:latest nest-server start
+
+The generic invocation command line for the ``nest-server`` command
+looks as follows::
+
+  nest-server <command> [-d] [-o] [-h <host>] [-p <port>] [-u <user>]
+
+Possible commands are `start`, `stop`, `status`, or `log`. The meaning
+of the other arguments is as follows:
+
+-d
+    Daemonize the server process
+-o
+    Print all output to both the console and the logger
+-h <host>
+    Use hostname/IP address <host> for the server [default: 127.0.0.1]
+-p <port>
+    Use port <port> for opening the socket [default: 5000]
+-u <uid>
+    Run the server under the user with ID <user>
+
+Run with MPI
+~~~~~~~~~~~~~~~~
+
+If NEST was compiled with support for :ref:`distributed computing via
+MPI <distributed_computing>`, it will usually execute the exact same
+simulation script on each of the MPI processes. With NEST Server, this
+would normally mean that one NEST Server instance would be spawned for
+each rank in a multi-process NEST simulation. To prevent this from
+happening, we provide a special version of the NEST Server command for
+use with MPI. It can be run as follows::
+
+  mpirun -np N nest-server-mpi [--host HOST] [--port PORT]
+
+If run like this, the RESTful API of the NEST Server will only be
+served by the MPI process with rank 0 (called the `master`), while all
+other N-1 ranks will start the NEST Server in `worker` mode. Upon
+receiving a request, the master relays all commands to the workers,
+which execute them, collect all result data, and send it back to the
+master. The master then receives and combines all worker responses,
+and replies to the caller of the NEST Server API.
+
+The response data in such a distributed scenario looks almost
+completely the same as one coming from the serial version of the NEST
+Server. The only difference may be that information pertaining to
+process-local data structures is being replaced by generic values.
+
+.. _nest_server_client:
+
+The NEST Server Client
+----------------------
+
+The easiest way to interact with the NEST Server is the `NEST Server
+Client` provided in ``examples/NESTServerClient`` in the source
```

Comment:
Should the `NESTServerClient `be in examples? the directory itself has an examples folder.
Also the file names of the examples could be improved:
`NESTClient_example.py`  and `NESTClient_script.py `

Reply:
IIRC @babsey originally had it in `extras/` and I voted for moving it to `examples`. As this is actually its own Python package with no dependencies on NEST whatsoever, I'd like to spin it off into its own `pip`-intallable repository mid-term to make it easier for others to build on it.

So in summary, I think we should just leave it as is for now.